### DISCUSSION TITLE
Remove  alarm code requierment

### DISCUSSION THREAD

**Discussion by SantiagoSotoC** [TD22057/ad-mqtt#13] [General]
Hello, I was thinking of removing the alarm code from the code or using a list of passwords to be able to use more than one user, what do you think?

**Comment by TD22057** [comment]
Sorry - not sure what that means.  The alarm code has to be in the input config or it has to sent via MQTT.  I didn't bother w/ the MQTT approach but it wouldn't be hard to add.  I don't know what you mean by a list of passwords - the alarm code is required to send to the alarmdecoder.

**Reply by SantiagoSotoC** [reply]
I mean that it should be allowed to use more than one password, using the one received by mqtt.
The reason for proposing this is to allow all panel users to be used

**Comment by TD22057** [comment]
IMO the way that should be done is to make the config shown here to be an option in the discovery system and the Bridge.  The question is where should codes be validated?  Enabling this option makes HASS send whatever code it receives to ad-mqtt.  Either ad-mqtt has to have a valid list of codes and check it or it has to pass the code to the alarm and see what happens.  
https://www.home-assistant.io/integrations/alarm_control_panel.mqtt/#configurations-with-remote-code-validation

The question is: what happens when you enter a wrong code?  With the current setup, HASS shows you it's wrong because it validates on the front end.  As far as I know, there isn't a way to send an error message back to HASS to show that the code was incorrect.
